src/dndbeyond/content-scripts/character.js

```diff
@@ -419,11 +420,13 @@ function rollItem(force_display = false, force_to_hit_only = false, force_damage
             if (character.getSetting("ranger-dread-ambusher", false)) {
                 damages.push("1d8");
                 damage_types.push("Ambush");
-                character.mergeCharacterSettings({ "ranger-dread-ambusher": false });
+                settings_to_change["ranger-dread-ambusher"] = false;
             }
-            if (character.hasClassFeature("Hunter’s Prey: Colossus Slayer")) {
+            if (character.hasClassFeature("Hunter’s Prey: Colossus Slayer") &&
+                character.getSetting("ranger-colossus-slayer", false)) {
                 damages.push("1d8");
                 damage_types.push("Colossus Slayer");
+                settings_to_change["ranger-colossus-slayer"] = false; // Reset to off since CS only applies once per turn
```

Comment:
CS shouldn't be resetting every time, it's supposed to be considered as Sneak Attack, where if you enable it, it would appear, but the player can decide to count it or not. Most likely, player will have one attack per turn, or only one of their two attacks will succeed. It's easier to ignore a number when doing the math, than to go into settings and re-enable it before each click.